### PR TITLE
Ensure that relational fields are not considered at index init time

### PR DESCRIPTION
**Pull request by johngriebel**
See issue #226 for more details
Signed-off-by: John Griebel <johnkgriebel@gmail.com>

### CODE REVIEW

**Comment by johngriebel** [comment]
@PLNech This seems to work on the various combinations of Python and Django in my testing. This change is implicitly tested by a bunch of different unit tests, but I did add a specific one as you asked. Let me know if you don't feel the test I added is sufficient, or if there are any conventions I need to follow.

File: algoliasearch_django/models.py [hunk 1]
```diff
@@ -80,10 +80,7 @@ def __init__(self, model, client, settings):
         self.__named_fields = {}
         self.__translate_fields = {}
 
-        try:
-            all_model_fields = model._meta.get_all_field_names()
-        except AttributeError:  # get_all_field_names is removed in Django >= 1.10
-            all_model_fields = [f.name for f in model._meta.get_fields()]
+        all_model_fields = [f.name for f in model._meta.get_fields() if not f.is_relation]
```
Comment (PLNech) [inline algoliasearch_django/models.py]: I would use `_meta.local_fields` instead on Django < 1.8 :

```diff
diff --git a/algoliasearch_django/models.py b/algoliasearch_django/models.py
index 46eae60..f9d7caf 100644
--- a/algoliasearch_django/models.py
+++ b/algoliasearch_django/models.py
@@ -80,7 +80,10 @@ class AlgoliaIndex(object):
         self.__named_fields = {}
         self.__translate_fields = {}

-        all_model_fields = [f.name for f in model._meta.get_fields() if not f.is_relation]
+        try:
+            all_model_fields = [f.name for f in model._meta.get_fields() if not f.is_relation]
+        except AttributeError: # get_fields requires Django >= 1.8
+            all_model_fields = [f.name for f in model._meta.local_fields]
```

Comment (johngriebel) [reply algoliasearch_django/models.py]: Hi @PLNech whoops, no I didn't run the tests using `tox`. I've never used it before so it totally slipped my mind when running the tests. I'll make these changes some time today! Thanks for the review.

Comment (PLNech) [reply algoliasearch_django/models.py]: Noprob, you're welcome!

Comment (PLNech) [reply algoliasearch_django/models.py]: Hey @johngriebel, do you want to move forward with this or should I apply these changes on top of your contribution and merge? 🙂

Comment (johngriebel) [reply algoliasearch_django/models.py]: @PLNech Hey, my apologies for letting this sit so long. My trial key with Algolia expired, then in the meantime my day job has gotten hectic; you know how it goes. Anyhow, go ahead and apply your changes on top of mine!

Comment (PLNech) [reply algoliasearch_django/models.py]: Hey, no need to apologise. I definitely can relate to this!

I'll go with these changes and merge then, thanks again for the report and contribution 👍

**Comment by PLNech** [comment]
Closing and reopening as #233 to add the discussed fix.

**Pull request closed without merging**